### DISCUSSION TITLE
常规播放器模式下鼠标滚轮调整视频音量

### DISCUSSION THREAD

**Discussion by pinaZC** [the1812/Bilibili-Evolved#4283] [功能建议 / Ideas]
目前官方默认设置是：常规视频模式下键盘的↑↓箭头调整音量，全屏及网页全屏可以用鼠标滚轮调整音量
这时候的问题就是如果视频音量不合适不能及时调节视频音量大小（虽然键盘已经很方便了，但是大家点视频还是靠鼠标的吧，去找键盘按键还是多了一步操作）
目前确实可以通过 默认播放器模式 这个组件设置成网页全屏来实现这个效果，但大部分情况下看视频还是会先以常规的页面开始看视频
所以 能不能实现鼠标在视频播放区域的时候，可以通过鼠标滚轮控制音量大小
（或者有没有像potplayer一样的 声音规格化 插件推荐 找了好久也没找到）
谢谢大佬

**Comment by Tinhone** [comment]
我个人也很需要类似声音规格化的插件，不过是平衡每个视频之间总体的音量（

**Comment by the1812** [comment]
如果能规格化就很不错, 但我对声音处理这块不熟

**Comment by WakelessSloth56** [comment]
响度规格化的处理很好实现，通过 [Web Audio API](https://developer.mozilla.org/en-US/docs/Web/API/Web_Audio_API) 拿到音频的 PCM 数据，然后用 ReplayGain 等算法。
不过得分析整个音频，对于在线的流媒体这里更不好实现吧…………
